internal/controllers/device_config_controller.go

```diff
@@ -294,34 +252,29 @@ func (d *DeviceConfigController) GetConfigURLs(c *fiber.Ctx) error {
 	vin := c.Params("vin")
 
 	// Set up gRPC call to retrieve UserDevice by VIN
-	req := &pb.GetUserDeviceByVINRequest{Vin: vin}
-	ud, err := d.usersClient.GetUserDeviceByVIN(context.Background(), req)
+	ud, err := d.userDeviceSvc.GetUserDeviceByVIN(context.Background(), vin)
 	if err != nil {
 		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
 			"error": fmt.Sprintf("Failed to retrieve user device for VIN: %s", vin),
 		})
 	}
+	// Query templates, filter by protocol and powertrain
```

Comment:
one thing is that currently we use the naming convention `CAN11_500` and `CAN29_500` but in user devices it will just return as strings `6` or `7` respectively, so you'll need a simple func to map that